### PR TITLE
Casminst 3955

### PR DESCRIPTION
**Pull request by phalseth-hpe**
These changes look for private repository credentials present in the Jenkins pipeline environment, and make them available for use in the rpm-index and rpm-sync processes when needing to access private repository resources during CSM packaging.

RELATED PRs: 
https://github.hpe.com/hpe/hpc-shastarelm-packaging-tools/pull/2

https://github.hpe.com/hpe/hpc-shastarelm-release/pull/9/files

### CODE REVIEW

File: release.sh [hunk 1]
```diff
@@ -57,6 +57,14 @@ RELEASE_VERSION_BUILDMETADATA="$(echo "$RELEASE_VERSION" | perl -pe "s/${semver_
...
 #
 
+#serialize an object containing repo credentials to disk, and put the path to it in an environment variable
+if [ ! -z "$ARTIFACTORY_USER" ] && [ ! -z "$ARTIFACTORY_TOKEN" ]; then
+    export REPOCREDSPATH="/tmp/"
+    export REPOCREDSFILENAME="repo_creds.json"
+    export REPOCREDSFULL=$REPOCREDSPATH$REPOCREDSFILENAME
+    jq --null-input   --arg url "sles-mirror" --arg realm "Artifactory Realm" --arg user "$ARTIFACTORY_USER"   --arg password "$ARTIFACTORY_TOKEN"   '{($url): {"realm": $realm, "user": $user, "password": $password}}' > $REPOCREDSFULL
+fi
+
```
Comment (mtupitsyn) [inline release.sh:67]: So it looks like `/tmp/repo_creds.json` will be left on jenkins runner? Is there any way to clean it up inevitably (e.g. even if Jnekins job gets cancelled)? Ideally we should not write sensitive information to filesystem, just pass username and password into docker container via environment variables.

File: Jenkinsfile.github [hunk 2]
```diff
@@ -81,14 +81,16 @@ pipeline {
...
                     steps {
-                        script {
-                            slackSend(channel: env.SLACK_CHANNEL_ALERTS, message: "<${env.BUILD_URL}|CSM ${env.RELEASE_VERSION}> - Running release.sh")
-                            sh """
-                                . build/.env/bin/activate
-                                make clean
-                                rm -fr dist
-                                ./release.sh
-                            """
+                        withCredentials([usernamePassword(credentialsId: credentialsId, usernameVariable: 'ARTIFACTORY_USER', passwordVariable: 'ARTIFACTORY_TOKEN')]) {
```
Comment (mtupitsyn) [inline Jenkinsfile.github:84]: Just a minor thing - I believe only `sh` call below needs to be encapsulated in `withCredentials`

File: release.sh (hunk 1 above)
Comment (phalseth-hpe) [reply release.sh:67]: I added code to Jenkinsfile.github to remove /tmp/repo_creds.json in the post release section, either success or fail

Comment (mtupitsyn) [reply release.sh:67]: I'm afraid this does not cover all cases (for example, aborted builds). Also hard to maintain, not clear where this file is getting created. IMO better just add 

    trap "rm -f '${REPOCREDSFULL}'" EXIT

after line 65 in `release.sh`.

**Review by mtupitsyn — APPROVED** [review]

**Review by heemstra — APPROVED** [review]

**Review by gbaker-hpe — APPROVED** [review]

**Pull request merged**